tls/s2n_certificate_keys.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include "api/s2n.h"
+
+struct s2n_certificate_key {
+    uint16_t public_key_libcrypto_nid;
+
+    /* modulus for RSA key, size for EC key */
+    uint16_t bits;
+};
+
+struct s2n_certificate_key_preferences {
+    uint8_t count;
+    const struct s2n_certificate_key *const *certificate_keys;
+};
+
+extern const struct s2n_certificate_key s2n_rsa_rsae_1024;
```

Comment:
Maybe I messed up some of the other header includes, but there are compile errors if I remove them, and we also use them in other places like this. 

I think they're necessary because the otherwise they conflict with the declaration in `s2n_certificate_keys.c`

Reply:
Ohh interesting ok. I thought extern was the default if left unspecified, [but apparently that's only the case for functions](https://stackoverflow.com/a/496476)!